### DISCUSSION TITLE
Upgrade Opencart v3.0.3.8 to Opencart 4.x

### DISCUSSION THREAD

**Discussion by testricha124** [opencart/opencart#12642] [Q&A]
Hi, Community

Upgrade Opencart runnig project v3.0.3.8 to Opencart 4.x

installed lots of 3rd party ocmod extension in my opencart 3 project 

upgrade is possible ??

**Comment by testricha124** [comment]
I tried upgrading opencart v3.0.3.8 to opencart 4.x my current project ,, but not succeed and arise so many errors,
and admin side 0 modules.

**Reply by IP-CAM** [reply]
Better install OC v.4.x separately, an Upgrade won't work.

**Comment by felipoantonoff** [comment]
I recommend waiting for a future version of Opencart 4.
But if you want to migrate, install Opencart 4 from scratch and reinstall the desired modules/apps for it and made for Opencart 4.

The Bank part makes a backup of the old store, normalizes the Bank to Opencart 4 and migrates to it, only the settings and some data that I do not recommend doing this.

In general, I recommend checking with a good professional, especially the bank migration from Opencart 3 to 4, as you will be able to keep orders, customers, products, images, information page and other data.

**Comment by testricha124** [comment]
Yes, Got it and Thanks

**Comment by karan982** [comment]
@testricha124 

In your case upgrade is not possible, its batter to setup separate opencart 4x in staging environment. I provide data migration service OC 3x to OC 4x. 
https://www.opencart.com/index.php?route=marketplace/extension/info&extension_id=45482

You can contact me at our support email **codinginspect@gmail.com**
